src/app/settings/views/Repositories/RepositoriesList/RepositoriesList.test.tsx

```diff
@@ -109,17 +110,31 @@ describe("RepositoriesList", () => {
         </MemoryRouter>
       </Provider>
     );
-    let row = wrapper.find("TableRow[data-testid='repository-row']").at(2);
-    expect(row.hasClass("is-active")).toBe(false);
-    // Click on the delete button:
-    wrapper.find("TableRow").at(3).find("Button").at(1).simulate("click");
-    row = wrapper.find("TableRow[data-testid='repository-row']").at(2);
-    expect(row.hasClass("is-active")).toBe(true);
+
+    expect(
+      screen.queryByRole("row", {
+        name: "secret_archive http://www.website.com No Edit Delete",
```

Comment:
This feels a little fragile. It might be better to add an aria-label to the row. Same for the other instances of `screen.queryByRole("row" ...` in this file.

Reply:
I'm not 100% sure on how to do this, I think we'd have to modify the MainTable React component, specifically MainTableRow, to take an aria-label as a prop

Reply:
You should just be able to add it to the [row's object](https://github.com/canonical/maas-ui/blob/main/src/app/settings/views/Repositories/RepositoriesList/RepositoriesList.tsx#L37) which gets passed in addition to the defined row props: https://github.com/canonical/react-components/blob/main/src/components/MainTable/MainTable.tsx#L232.

Reply:
I've tried that, but if I use "aria-label", the code thinks it's an arithmetic operation (which makes sense because it's using a `-` in an object and not an HTML tag), and if I use `ariaLabel`, I get the following error:

`Warning: Invalid ARIA attribute `ariaLabel`. Did you mean `aria-label`?`

Which makes me think that the only remaining way to make it work is to modify that react component

Reply:
Ignore that, I forgot you can add props with the name wrapped in quotation marks :joy: :man_facepalming: